lib/schema.json

```diff
@@ -0,0 +1,167 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "type": "object",
+  "properties": {
+    "nodes": {
+      "type": "array",
+      "items": {
+        "type": "object",
+        "properties": {
+          "opcode": {
+            "type": "string",
+            "enum": [
```

Comment:
I'd say that the spec defines some specific opcodes that SHOULD be used, but any other opcode might be used as well.